### PR TITLE
Inheritance started but organization random generation isn't finished…

### PR DESCRIPTION
**Pull request by csizmadiazsolt**
Please take a look at the code. It's not ready yet, but mistakes may occur already... :D

### CODE REVIEW

File: Inheritance/src/main/java/inheritance/Employee.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package inheritance;
+
+public abstract class Employee {
+	protected String name;
+	protected Double salary;
+	
+	protected static int MAXNUMBEROFCEOS = 1;
```
Comment (ZsoltFabok) [inline Inheritance/src/main/java/inheritance/Employee.java]: These fields are not important. The numbers in the exercise are just samples for creating objects.

File: Inheritance/src/main/java/inheritance/EmployeeFactory.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package inheritance;
+
+public class EmployeeFactory {
+	
+	private static int MAXNUMOFCEOS = 1;
```
Comment (ZsoltFabok) [inline Inheritance/src/main/java/inheritance/EmployeeFactory.java]: Same as above

File: Inheritance/src/main/java/inheritance/EmployeeFactory.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package inheritance;
+
+public class EmployeeFactory {
```
Comment (ZsoltFabok) [inline Inheritance/src/main/java/inheritance/EmployeeFactory.java]: Nice :-)

File: Inheritance/src/main/java/inheritance/EmployeeFactory.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+			MAXNUMOFSECRETARIES--;
+			return new Secretary(name, salary);
+		}
+		
+		return null;
+	}
+	
+	public Employee getEmployee(EmployeeTypes empType, String name, Double salary, DeveloperType dType) {
+		if(empType == null) {
+			return null;
```
Comment (ZsoltFabok) [inline Inheritance/src/main/java/inheritance/EmployeeFactory.java]: It is not a good practice to return `null`. We didn't talk about it, so no worries.

File: Inheritance/src/main/java/inheritance/OrganizationGenerator.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	public List<Developer> developers = new ArrayList<Developer>();
+
+	public void generateOrganization() {
+		int numberOfManagers = generateRandomInRange(1,Employee.MAXNUMBEROFMANAGERS);
+		int numberOfSecretaries = generateRandomInRange(1,Employee.MAXNUMBEROFSECRETARIES);
+		int numberOfDevelopers = generateRandomInRange(1, Employee.MAXNUMBEROFDEVELOPERS);
+		
+		ceo = (CEO) factory.getEmployee(EmployeeTypes.CEO, "Mr Bossman", 1000000.0);
+		
+		for(int i=0; i<numberOfManagers; i++) {
```
Comment (ZsoltFabok) [inline Inheritance/src/main/java/inheritance/OrganizationGenerator.java]: Can you do this with java8? Just curious.

**Comment by ZsoltFabok** [comment]
We shoud talk about this exercise. Regardless of the outcome, I'm missing the tests and sonar found some issues.

File: Inheritance/src/main/java/inheritance/Employee.java (hunk 1 above)
Comment (csizmadiazsolt) [inline Inheritance/src/main/java/inheritance/Employee.java]: Replaced according to our conversation.

File: Inheritance/src/main/java/inheritance/EmployeeFactory.java (hunk 2 above)
Comment (csizmadiazsolt) [inline Inheritance/src/main/java/inheritance/EmployeeFactory.java]: Replaced according to our conversation.

File: Inheritance/src/main/java/inheritance/EmployeeFactory.java (hunk 4 above)
Comment (csizmadiazsolt) [inline Inheritance/src/main/java/inheritance/EmployeeFactory.java]: Discuss it on Monday.

**Pull request closed without merging**